src/test/java/at/ac/tuwien/kr/alpha/grounder/RuleGroundingOrderTest.java

```diff
@@ -123,16 +125,20 @@ public void testToString_longerSimpleNonGround() {
 
 	@Test
 	public void testPositionFromWhichAllVarsAreBound_joinedNonGround() {
-		Program program = parser.parse("a(X) :- b(X), c(X,Y), d(X,Z), not e(X).");
+		Program program = PROGRAM_PARSER.parse("a(X) :- b(X), c(X,Y), d(X,Z), not e(X).");
 		RuleGroundingOrders rgo0 = computeGroundingOrdersForRule(program, 0);
-		assertTrue(2 <= rgo0.orderStartingFrom(literal("b", "X")).getPositionFromWhichAllVarsAreBound());
-		assertTrue(1 <= rgo0.orderStartingFrom(literal("c", "X", "Y")).getPositionFromWhichAllVarsAreBound());
-		assertTrue(1 <= rgo0.orderStartingFrom(literal("d", "X", "Z")).getPositionFromWhichAllVarsAreBound());
+		final Literal litBX = PROGRAM_PART_PARSER.parseLiteral("b(X)");
+		final Literal litCXY = PROGRAM_PART_PARSER.parseLiteral("c(X,Y)");
+		final Literal litDXZ = PROGRAM_PART_PARSER.parseLiteral("d(X,Z)");
+		assertTrue(2 <= rgo0.orderStartingFrom(litBX).getPositionFromWhichAllVarsAreBound());
+		assertTrue(1 <= rgo0.orderStartingFrom(litCXY).getPositionFromWhichAllVarsAreBound());
+		assertTrue(1 <= rgo0.orderStartingFrom(litDXZ).getPositionFromWhichAllVarsAreBound());
 	}
 
 	private RuleGroundingOrders computeGroundingOrdersForRule(Program program, int ruleIndex) {
 		Rule rule = program.getRules().get(ruleIndex);
-		NonGroundRule nonGroundRule = NonGroundRule.constructNonGroundRule(rule);
+		final NonGroundRule nonGroundRule1 = NonGroundRule.constructNonGroundRule(rule);
+		NonGroundRule nonGroundRule = nonGroundRule1;
```

Comment:
I do not fully see the reason for this duplicate variable. If it is necessary, please add a comment explaining the reason.

Reply:
I also don´t see a reason.